src/foam/nanos/auth/email/EmailTokenService.js

```diff
@@ -34,9 +34,8 @@ foam.CLASS({
 `try {
 DAO tokenDAO = (DAO) getX().get("localTokenDAO");
 DAO userDAO  = (DAO) getX().get("localUserDAO");
-AppConfig appConfig = (AppConfig) getX().get("appConfig");
-String url = appConfig.getUrl()
-    .replaceAll("/$", "");
+AppConfig appConfig = user.findGroup(x).getAppConfig(x);
```

Comment:
```suggestion
AppConfig appConfig = (AppConfig) x.get("appConfig");
```

Does this work instead? The app config should already be set based on the group (if there's a group available). This happens in `Session.applyTo`, which gets called for every request to create the user's server-side request context. Not sure why this was using `getX()` instead of `x` here.

Reply:
The AppConfig is set based on the group when the user is logged in. But when the user is not logged in we still need to go through the user's group to get group appconfig overrides - such as sme url. 
yes, not sure about getx().

Reply:
Ah, thanks, that makes sense. And I guess the user in `x` when there is one isn't necessarily the one we want to use anyway, it makes more sense to use the one passed as an argument to `generateTokenWithParameters` because that's who's getting the email.